apps/roam/src/utils/pageToMarkdown.ts

```diff
@@ -0,0 +1,347 @@
+import getPageMetadata from "./getPageMetadata";
+import getPageViewType from "roamjs-components/queries/getPageViewType";
+import type { Result } from "roamjs-components/types/query-builder";
+import type { DiscourseNode } from "./getDiscourseNodes";
+import type { PullBlock, TreeNode, ViewType } from "roamjs-components/types";
+import getFullTreeByParentUid from "roamjs-components/queries/getFullTreeByParentUid";
+import getDiscourseContextResults from "./getDiscourseContextResults";
+import isFlagEnabled from "./isFlagEnabled";
+import XRegExp from "xregexp";
+import {
+  findReferencedNodeInText,
+  getReferencedNodeInFormat,
+} from "./formatUtils";
+import {
+  getFilename,
+  toLink,
+  pullBlockToTreeNode,
+  collectUids,
+} from "./exportUtils";
+
+const MATCHES_NONE = /$.+^/;
+
+// Roam embed syntax: {{[[embed]]: ((block-uid)) }}
+// Roam embed syntax: {{[[embed-path]]: ((block-uid)) }}
+// Also handles multiple parentheses: {{[[embed]]: ((((block-uid)))) }}
+const EMBED_REGEX =
+  /{{\[\[(?:embed|embed-path)\]\]:\s*\(\(+\s*([\w\d-]{9,10})\s*\)\)+\s*}}/;
+
+// Roam embed-children syntax: {{[[embed-children]]: ((block-uid)) }}
+const EMBED_CHILDREN_REGEX =
+  /{{\[\[embed-children\]\]:\s*\(\(+\s*([\w\d-]{9,10})\s*\)\)+\s*}}/;
+
+const viewTypeToPrefix = {
+  bullet: "- ",
+  document: "",
+  numbered: "1. ",
+};
+
+const handleDiscourseContext = async ({
+  includeDiscourseContext,
+  uid,
+  pageTitle,
+  appendRefNodeContext,
+}: {
+  includeDiscourseContext: boolean;
+  uid: string;
+  pageTitle: string;
+  appendRefNodeContext: boolean;
+}) => {
+  if (!includeDiscourseContext) return [];
+
+  const discourseResults = await getDiscourseContextResults({
+    uid,
+  });
+  if (!appendRefNodeContext) return discourseResults;
+
+  const referencedDiscourseNode = getReferencedNodeInFormat({ uid });
+  if (referencedDiscourseNode) {
+    const referencedResult = findReferencedNodeInText({
+      text: pageTitle,
+      discourseNode: referencedDiscourseNode,
+    });
+    if (!referencedResult) return discourseResults;
+    const appendedContext = {
+      label: referencedDiscourseNode.text,
+      results: { [referencedResult.uid]: referencedResult },
+    };
+    return [...discourseResults, appendedContext];
+  }
+
+  return discourseResults;
+};
+
+const handleFrontmatter = ({
+  frontmatter,
+  rest,
+  result,
+}: {
+  frontmatter: string[];
+  rest: Record<string, unknown>;
+  result: Result;
+}) => {
+  const yaml = frontmatter.length
+    ? frontmatter
+    : [
+        "title: {text}",
+        `url: https://roamresearch.com/#/app/${window.roamAlphaAPI.graph.name}/page/{uid}`,
+        `author: {author}`,
+        "date: {date}",
+      ];
+  const resultCols = Object.keys(rest).filter((k) => !k.includes("uid"));
+  const yamlLines = yaml.concat(resultCols.map((k) => `${k}: {${k}}`));
+  const content = yamlLines
+    .map((s) =>
+      s.replace(/{([^}]+)}/g, (_, capt: string) => {
```

Comment:
Am I to assume that this page was copied over 1:1 with no changes, which is why the es lints are here?  Next time just make a quick inline github comment at the top of the page so there is no ambiguity.

Reply:
Lets stick to keeping it in the body of the PR.  And keep the loom video specific to a single PR.  That way it can be focused.  But walking through and adding inline GitHub comments is usually a really good idea.  How about let's commit to try adding a few next PR?